ushahidi/platform-client
# Post edit pane WIP

This pull request makes the following changes:
- Adds edit-functionality to split-view
- Shows correct toolbar-button based on where in application you are (based on currentView)
- Fixes edit-button in post-action dropdown
- Adds message with nb of posts out of total posts on top of post-list
LEFT TO DO!
~- Check error-messages~
~- Add message to user that a post is not changed when clicking save and no changes were made~
- Make edit-button in post-action-dropdown in detail-view work
- Display 'leave-form-warning' if clicking outside of the edit-form
- Update post-cards and detailed view with new details when a change has been saved
- When saving a post that doesn't match the current filtering, the post should be removed from the list and the detailed view of the next one should appear
- General style-fixes


Testing checklist:
- Go to data view and click on a post + edit-button
- [ ] Edit-view should appear
- Don't make a change and click save
- [ ] A modal with the text "You have not made any changes" should appear 
- Make a change and click save
- [ ] Changes should be saved and post-card to the left should update
- Make a change and click cancel, a warning should appear with the option to navigate away anyway or save
- [ ] Clicking 'navigate away' should leave the edit-mode and not save changes
- [ ] Clicking 'save' should save changes and leave the edit mode 
- Go to map-view and click on a post + select edit-post on the post-action-dropdown
- [ ] the 'old' edit-view should be visible




Fixes ushahidi/platform# .

Ping @ushahidi/platform
